src/test/shared/utilities/workspaceUtils.test.ts

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as assert from 'assert'
+import * as path from 'path'
+import * as sinon from 'sinon'
+import * as vscode from 'vscode'
+import { findParentProjectFile } from '../../../shared/utilities/workspaceUtils'
+
+describe('findParentProjectFile', async () => {
+    const workspaceDir = path.join('code', 'workspace')
+    before(() => {
+        const sandbox = sinon.createSandbox()
+        sandbox.stub(vscode.workspace, 'getWorkspaceFolder').returns({
+            name: 'tempFolder',
+            index: 0,
+            uri: vscode.Uri.parse(workspaceDir),
+        })
+    })
+    const sourceCodeUri = vscode.Uri.file(path.join(workspaceDir, 'someproject', 'src', 'Program.cs'))
+    const projectInSameFolderUri = vscode.Uri.file(path.join(workspaceDir, 'someproject', 'src', 'App.csproj'))
+    const projectInParentFolderUri = vscode.Uri.file(path.join(workspaceDir, 'someproject', 'App.csproj'))
+    const projectInParentParentFolderUri = vscode.Uri.file(path.join(workspaceDir, 'App.csproj'))
+    const projectInParentButOutOfWorkspace = vscode.Uri.file(path.join('code', 'App.csproj'))
+    const projectOutOfParentChainUri = vscode.Uri.file(path.join('code', 'someotherproject', 'App.csproj'))
```

Comment:
no need for path.join() here

Reply:
They're necessary with the move to integration tests.